pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -744,6 +749,11 @@ func PodFitsResources(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 
 	for rName, rQuant := range podRequest.ScalarResources {
 		if allocatable.ScalarResources[rName] < rQuant+nodeInfo.RequestedResource().ScalarResources[rName] {
+			// Ignore missing extended resources
+			// This is useful in scenarios where predicate checks are run on nodes with pods requesting cluster level resources which cannot be found on nodes.
+			if _, exists := nodeInfo.RequestedResource().ScalarResources[rName]; !exists && (ignoreMissingExtendedResources && v1helper.IsExtendedResourceName(rName)) {
```

Comment:
will `allocatable.ScalarResources` have a cluster level resource?

Reply:
May be i am reading code incorrectly. Should not `rName` be checked in `allocatable` to verify existence?